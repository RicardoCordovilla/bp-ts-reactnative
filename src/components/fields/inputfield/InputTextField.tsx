import React, { useState } from 'react'
import { Text, TextInput, View } from 'react-native'
import styles from './inputtextfield.styles'
import { InputFieldProps } from '../../../types/types'


const InputTextField: React.FC<InputFieldProps> = ({ label, placeholder, validation, disabled, value }) => {

    const [text, setText] = useState<string>('')
    const [error, setError] = useState<string | null>('')

    const validationTypes: Record<string, { regex: string, error: string }> = {
        number: {
            regex: '^[0-9]*$',
            error: 'Id inválido, solo números'
        },
        text: {
            regex: '^[a-zA-Z]*$',
            error: 'Solo texto'
        },
        date: {
            regex: '^[0-9]{4}-[0-9]{2}-[0-9]{2}$',
            error: 'Formato de fecha incorrecto (yyyy-mm-dd)'
        }
    }

    const handleChange = (text: string) => {
        if (validation) {
            const regex = new RegExp(validationTypes[validation].regex)
            if (!regex.test(text)) {
                setError(validationTypes[validation].error)
            } else {
                setError(null)
            }
        }
        // onChange(text)
        setText(text)
    }

    return (
        <View style={styles.field_container}>
            <Text style={styles.field_label}>{label}</Text>
            <TextInput
                style={styles.field_input}
                placeholder={placeholder}
                onChangeText={handleChange}
                value={text}
                editable={!disabled}
            />
        </View>
    )
}

export default InputTextField
