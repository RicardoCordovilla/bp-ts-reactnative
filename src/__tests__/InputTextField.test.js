import React from 'react';
import renderer from 'react-test-renderer';
import InputTextField from './../../src/components/fields/inputfield/InputTextField';


// test('renders InputTextField correctly', () => {
//     const component = renderer.create(<InputTextField />);
//     const tree = component.toJSON();
//     expect(tree).toMatchSnapshot();
// });

describe('InputTextField', () => {
    it('renders correctly', () => {
        const tree = renderer.create(
            <InputTextField
                label="Test Label"
                value="Test Value"
                placeholder="Test Placeholder"
                onChange={() => {}}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
