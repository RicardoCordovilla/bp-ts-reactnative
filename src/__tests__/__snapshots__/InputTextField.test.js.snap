// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputTextField renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "flexDirection": "column",
      "gap": 5,
      "height": "auto",
      "marginHorizontal": 10,
      "marginVertical": 10,
    }
  }
>
  <Text
    style={
      {
        "color": "#346",
        "fontSize": 16,
        "marginBottom": 10,
      }
    }
  >
    Test Input
  </Text>
  <TextInput
    editable={true}
    onChangeText={[Function]}
    style={
      {
        "backgroundColor": "#fff",
        "borderColor": "#ccc",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#666",
        "fontSize": 16,
        "height": 60,
        "padding": 10,
        "paddingVertical": 15,
      }
    }
    value=""
  />
</View>
`;

exports[`renders InputTextField correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#fff",
      "flexDirection": "column",
      "gap": 5,
      "height": "auto",
      "marginHorizontal": 10,
      "marginVertical": 10,
    }
  }
>
  <Text
    style={
      {
        "color": "#346",
        "fontSize": 16,
        "marginBottom": 10,
      }
    }
  />
  <TextInput
    editable={true}
    onChangeText={[Function]}
    style={
      {
        "backgroundColor": "#fff",
        "borderColor": "#ccc",
        "borderRadius": 5,
        "borderWidth": 1,
        "color": "#666",
        "fontSize": 16,
        "height": 60,
        "padding": 10,
        "paddingVertical": 15,
      }
    }
    value=""
  />
</View>
`;
